crates/database/interface/src/async_db.rs

```diff
@@ -48,7 +48,7 @@ pub trait DatabaseAsync {
 /// Use [WrapDatabaseAsync] to provide [DatabaseRef] implementation for a type that only implements this trait.
 pub trait DatabaseAsyncRef {
     /// The database error type.
-    type Error: Send;
+    type Error: core::error::Error + Send;
```

Comment:
I would consider adding a `'static` bound here, as described in https://github.com/bluealloy/revm/pull/1840/files#r1819000225